packages/service-library/tests/test_async_utils.py

```diff
@@ -0,0 +1,150 @@
+import asyncio
+import copy
+import random
+from collections import deque
+from time import time
+from typing import Any, Dict, List
+
+import pytest
+
+from servicelib.async_utils import run_serialized_in_context
+
+
+class LockedStore:
+    __slots__ = ("_queue", "_lock")
+
+    def __init__(self):
+        self._queue = deque()
+        self._lock = asyncio.Lock()
+
+    async def push(self, item: Any):
+        async with self._lock:
+            self._queue.append(item)
+
+    async def get_all(self) -> List[Any]:
+        async with self._lock:
+            return list(self._queue)
+
+
+async def test_context_aware_dispatch() -> None:
+    @run_serialized_in_context(target_args=["c1", "c3", "c3"])
```

Comment:
what is the idea with having twice "c3"?

Reply:
that was an error which was corrected